### PR TITLE
Develop merging to Master

### PR DESCRIPTION
**Pull request by theanhgen**

### CODE REVIEW

File: ups_modul/inout/inout.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
-from anthill import *
+from anthill import anthill, anthill_card, anthill_name, anthill_list
 from inout_gsheet import *
```
Comment (nguyeho7) [inline ups_modul/inout/inout.py]: what do you import from inout_gsheet?

File: ups_modul/inout/inout_gsheet.py [hunk 2]
```diff
@@ -3,178 +3,235 @@
...
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+4)
-        a2_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+6)
-        wks.update_cell(anthill_name[keys].row, days_range+7 ,"=ROUNDDOWN({}+{};0)".format(a1_notion, a2_notion))
-
-# writing the user time in the user cell in the day IN
-def user_time_log(day_in, user_name, hours, wks):
+def names_rows(anthill, wks):
+    '''
+    names from the anthill list in rows
```
Comment (nguyeho7) [inline ups_modul/inout/inout_gsheet.py]: Function header not clear

File: ups_modul/inout/inout_gsheet.py [hunk 3]
```diff
@@ -3,178 +3,235 @@
...
+    days_list = []
+    a1_notion, a2_notion = a_notion(1, 2, 1, days+1)
+    cell_list = wks.range("{}:{}".format(a1_notion, a2_notion))
+    for x in range(1, days+1):
+        days_list.append(x)
+    batch_cells(days_list, cell_list, wks)
+
+def header_calculation(days, wks):
+    '''
+    header labels for calculation
```
Comment (nguyeho7) [inline ups_modul/inout/inout_gsheet.py]: Function header not clear

File: ups_modul/inout/inout_gsheet.py [hunk 4]
```diff
@@ -3,178 +3,235 @@
...
-    SUM_tips_ratio(days_range, wks)
-    SUM_tips(days_range, wks)
-    SUM_total(days_range, wks)
-
-# botoms labels
-def labels_bottom(wks):
-    hours_label(wks)
-    SUM_day(days_range, wks)
-    SUM_labels(days_range, wks)
+def try_this_month():
```
Comment (nguyeho7) [inline ups_modul/inout/inout_gsheet.py]: This code will crash when the exception happens.

wks is created only in the try branch, not the exception handling branch

File: ups_modul/inout/inout.py (hunk 1 above)
Comment (theanhgen) [reply ups_modul/inout/inout.py]: try_this_month, update_worksheet, user_sheet_log

File: ups_modul/inout/inout_gsheet.py (hunk 2 above)
Comment (theanhgen) [reply ups_modul/inout/inout_gsheet.py]: names in anthill_list is written in first column sorted alphabeticly

File: ups_modul/inout/inout_gsheet.py (hunk 3 above)
Comment (theanhgen) [reply ups_modul/inout/inout_gsheet.py]: labels for the columns in the SUM part of the worksheets

File: ups_modul/inout/inout_gsheet.py (hunk 4 above)
Comment (theanhgen) [reply ups_modul/inout/inout_gsheet.py]: wks = auth_log(this_month)

is added to exception part

File: ups_modul/inout/inout_gsheet.py (hunk 3 above)
Comment (nguyeho7) [reply ups_modul/inout/inout_gsheet.py]: Try using verbs.

header_days():
'''
Writes the days of the month in the header row of the spreadsheet

'''

**Pull request merged**